### PR TITLE
test: add fuzz targets for the archive reader, redaction, and patch application

### LINKED ISSUE #254: test: add fuzz targets for the archive reader, redaction, and patch application
## Problem

**No fuzzing anywhere in the repo** — no `func Fuzz*`, no `testdata/fuzz/`,
no `testing/quick`, no property-based tests.

### 1. `FuzzOpenArchive`

Seed with the two existing goldens
(`internal/archive/testdata/golden-v1.kshrk`,
`golden-v1-passphrase.kshrk`). Assert no panic and no unbounded allocation.
This is the untrusted-input surface — `.kshrk` files move between
organizations.

### 2. `FuzzApplyRules` — `internal/redact/fields.go`

Over `tokenizePath` and `ApplyRules`. A panic here aborts a redaction run
**mid-way**, which for a redaction tool means a partially-redacted archive
that a user may believe is clean.

### 3. `FuzzApplyPatch` — `internal/server/writes.go:1193`

`applyPatch` is at 54.5%; the JSON-patch (RFC 6902) and `apply-patch+yaml`
branches are the uncovered ones.

## Also worth covering as ordinary tests

From the edge-case sweep, these have no coverage today:

- truncated **encrypted** file (distinct code path from tampering, which
  *is* tested)
- malformed JSON inside a record
- unicode / very long resource names — mitigated by `pathDir()` SHA-256
  hashing, but index keys and table printers handle them raw
- zip-slip in entry names — not currently exploitable, but no test pins the
  invariant

## Acceptance criteria

- [ ] Three fuzz targets committed with seed corpora
- [ ] A short fuzz run wired into CI (or a nightly)
- [ ] Any crashers found are fixed and their inputs added to the corpus

### PR DESCRIPTION
**Pull request by phenixblue**
## Summary
No fuzzing existed anywhere in the repo — no `func Fuzz*`, no `testdata/fuzz/`, no property-based tests. Adds the three targets in value order from #254:

- **`FuzzOpenArchive`** (`internal/archive`) — the untrusted-input surface: `.kshrk` files move between organizations. Seeded with the golden archives (plain and passphrase-encrypted, both format versions). Asserts `Open` never panics on arbitrary bytes, only ever errors.
- **`FuzzApplyRules`** (`internal/redact`) — exercises `tokenizePath` (via `ApplyRules`' per-rule path resolution) and `ApplyRules`' traversal together. A panic here aborts a redaction run mid-way — for a redaction tool that means a partially-redacted archive a user may believe is clean.
- **`FuzzApplyPatch`** (`internal/server`) — the write path's four patch strategies (JSON Patch, strategic-merge, `apply-patch+yaml`, plain merge), all of which decode an untrusted client-supplied body. `applyPatch` is at 54.5% coverage with the JSON-patch and `apply-patch+yaml` branches uncovered.

Adds `.github/workflows/fuzz.yml`, running each target for 30s on PRs/pushes touching the fuzzed packages — a future crasher gets committed to `testdata/fuzz/<target>/` and gates CI from then on.

The "also worth covering as ordinary tests" list in #254 (truncated encrypted file, malformed JSON in a record, unicode/long resource names, zip-slip) is explicitly separate from this issue's acceptance criteria and isn't addressed here.

Closes #254

## Test plan
- [x] `make fmt` / `go build ./...` / `go vet ./...` / `go mod tidy` (no diff)
- [x] `go test ./...`
- [x] Ran each target locally for 25-30s (kind/Docker unavailable in this environment, so no corpus could be seeded from a live capture):
  - `FuzzOpenArchive`: completed, no crash
  - `FuzzApplyRules`: ~4M execs, no crash
  - `FuzzApplyPatch`: ~860k execs, no crash
- [ ] `.github/workflows/fuzz.yml` will run for real on this PR (it's scoped to trigger on changes under `internal/archive/**`, `internal/redact/**`, `internal/server/**`, and the workflow file itself)

### CODE REVIEW

File: internal/server/fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		{``, ``, "application/json-patch+json", "", "v1", "pods"},
+		{`not json`, `not json either`, "application/strategic-merge-patch+json", "", "v1", "pods"},
+	}
+	for _, s := range seeds {
+		f.Add([]byte(s.current), []byte(s.patch), s.contentType, s.group, s.version, s.resource)
+	}
+
+	f.Fuzz(func(t *testing.T, current, patch []byte, contentType, group, version, resource string) {
+		_, _ = applyPatch(current, patch, contentType, group, version, resource)
+	})
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/server/fuzz_test.go:53]: Good catch. Constrained `contentType` to `supportedPatchContentTypes` (the four real values, selected via a fuzzed index, with an optional `; charset=utf-8` parameter) instead of a free-form string, and dropped the empty-content-type seed. The fuzzer's budget now goes toward current/patch/GVR mutations within `applyPatch`'s actual contract instead of Content-Type values production never reaches. Ran ~500k execs afterward, no crashers. Fixed in 32b7669.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
